--- a/src/backend/drm/DRM.cpp
+++ b/src/backend/drm/DRM.cpp
@@ -49,6 +49,9 @@
 #define SP CSharedPointer
 
 static bool shouldSubmitCTM(SP<SDRMConnector> connector, const COutputState::SInternalState& state, bool modeset) {
+    if (!connector->crtc || !connector->crtc->props.values.ctm)
+        return false;
+
     if (state.committed & COutputState::AQ_OUTPUT_STATE_CTM)
         return true;
 
@@ -875,7 +878,7 @@
         return false;
     }
 
-    rendererState.renderer = CDRMRenderer::attempt(backend.lock(), gpu->renderNodeFd >= 0 ? gpu->renderNodeFd : gpu->fd);
+    rendererState.renderer = CDRMRenderer::attempt(backend.lock(), newAllocator);
 
     if (!rendererState.renderer) {
         backend->log(AQ_LOG_ERROR, "drm: initMgpu: no renderer");
@@ -1541,7 +1544,7 @@
         if (!a)
             backend->log(AQ_LOG_ERROR, "drm: onReady: no renderer for gl formats");
         else {
-            auto r = CDRMRenderer::attempt(backend.lock(), gpu->renderNodeFd >= 0 ? gpu->renderNodeFd : gpu->fd);
+            auto r = CDRMRenderer::attempt(backend.lock(), a);
             if (!r)
                 backend->log(AQ_LOG_ERROR, "drm: onReady: no renderer for gl formats");
             else {
